barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.hpp

```diff
@@ -336,39 +381,104 @@ template <class ProverInstances_> class ProtoGalaxyProver_ {
 
             for (size_t idx = start; idx < end; idx++) {
                 // No need to initialise extended_univariates to 0, it's assigned to
-                extend_univariates(extended_univariates[thread_idx], instances, idx);
+                if constexpr (disable_optimisation) {
+                    extend_univariates(extended_univariates[thread_idx], instances, idx);
```

Comment:
you should probably be able to get the correct function at compile time with overloading or concepts rather than the if `constexpr` clauses

Reply:
but maybe i'm wrong and you tried before

Reply:
Arguments don't change, so how would I overload it?

Reply:
Oh, then it would be 2 different implementations, if I just decided to make this a concept. I'm not sure that's better here. I want the code to be reused